src/lib.rs

```diff
@@ -42,38 +40,115 @@ use treesitter::maybe_get_query;
 use use_printer::get_printer;
 use use_searcher::get_searcher;
 
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("couldn't read query file {path_to_query_file:?}")]
+    QueryFileReadError {
+        path_to_query_file: PathBuf,
+        source: io::Error,
+    },
+    #[error("{}",
+        match .0.len() {
+            1 => {
+                let (supported_language, query_error) = &.0[0];
+                format!("couldn't parse query for {supported_language:?}: {query_error}")
+            }
+            _ => {
+                let mut attempted_parsings = .0
+                    .iter()
+                    .map(|(supported_language, _)| format!("{supported_language:?}"))
+                    .collect::<Vec<_>>();
+                attempted_parsings.sort();
+                format!(
+                    "couldn't parse query for {}",
+                    join_with_or(&attempted_parsings)
+                )
+            }
+        }
```

Comment:
Ha, I don't think I've ever seen so much code *inside* an attribute before.

Reply:
Ya I initially had them separated as two different `Error` variants (ie the "tried parsing in multiple languages" vs "only tried parsing in a single language" cases) but really that's conceptually the same error, it's just an error-formatting choice to only show the query error in the tried-single-language case, so does in fact seem like it belongs here